src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -237,10 +239,12 @@ protected override IEnumerable<TypeInferenceInfo> InferTypesWorker_DoNotCallDire
                     PostfixUnaryExpressionSyntax postfixUnary => InferTypeInPostfixUnaryExpression(postfixUnary, token),
                     PrefixUnaryExpressionSyntax prefixUnary => InferTypeInPrefixUnaryExpression(prefixUnary, token),
                     ReturnStatementSyntax returnStatement => InferTypeForReturnStatement(returnStatement, token),
+                    SingleVariableDesignationSyntax singleVariableDesignationSyntax => InferTypeForSingleVariableDesignation(singleVariableDesignationSyntax),
                     SwitchLabelSyntax switchLabel => InferTypeInSwitchLabel(switchLabel, token),
                     SwitchStatementSyntax switchStatement => InferTypeInSwitchStatement(switchStatement, token),
                     ThrowStatementSyntax throwStatement => InferTypeInThrowStatement(throwStatement, token),
                     TupleExpressionSyntax tupleExpression => InferTypeInTupleExpression(tupleExpression, token),
+                    UnaryPatternSyntax unaryPattern => GetPatternTypes(unaryPattern),
```

Comment:
are all tehse new cases tested?

Reply:
note: we have explicit tests around the TypeInferenceService, can you updat there?

Reply:
I copied the tests from EnumCompletionProviderTests to TypeInferrerTests in 3479b73e0bc76de30601aa547efa66e7f2ae2584, 48ec7934328121de6d761535c4822c7d0b7b9df6 and de23198eeebeb37ca3ed072d48acab99948c6c5a

TestEnumInPatterns_Is_ConstUnaryAndBinaryPattern

https://github.com/dotnet/roslyn/pull/49632/files#diff-9cda451529a635b3dfcee6e983569f6139cdb5be2c64d916dca49bdb37152a2aR3109

> are all tehse new cases tested?

Yes. All code paths are covered by the new tests in TypeInferrerTests.

Tests are duplicated in EnumCompletionProviderTests and TypeInferrerTests with some minor differences now. I think this is valuable, but I can remove the tests from EnumCompletionProviderTests if you like.